qiskit/ignis/verification/randomized_benchmarking/circuits.py

```diff
@@ -143,7 +143,8 @@ def load_tables(max_nrb=2):
 def randomized_benchmarking_seq(nseeds=1, length_vector=None,
                                 rb_pattern=None,
                                 length_multiplier=1, seed_offset=0,
-                                align_cliffs=False):
+                                align_cliffs=False,
+                                interleaved_gates=None):
```

Comment:
These are going to get passed in as Clifford objects

Reply:
I think that it may be more difficult, since the function `compose_gates` in `clifford_utils.py` can only add gates to a Clifford object from a list of gates. In the current basic `Clifford.py` there is no function to compose two Clifford objects. 
I also think that for the users it may be easier to produce a list of gates than a Clifford object.

Reply:
Looking at the code 
1. `compose_gates` needs to be part of the clifford class
2. the clifford objects should be composable (this should be straightforward, let's ask @chriseclectic who understands the representation)

Then users input based on a clifford object (this way we know it's a clifford). The user can build that Clifford with the same list of gates so it's not really more work. The actual set of gates will get selected using the lookup table; that way it's in the same basis.

Reply:
The `Clifford.py` class should be a temporary class until there will be a Clifford python class in Terra, that will call the Aer Clifford simulator, so it includes only basic functionality.

As I remember, it was @chriseclectic who recommended which functions should be in the Clifford class, and which functions should be in `clifford_utils.py`  (based on the planned Clifford class).

If the `interleaved_gates` should be passed in as Clifford objects there are two possibilities: 

1. Add a function to the Clifford class that compose Clifford objects (is it basically just a multiplication of the two Symplectic matrices?)
**or:** 
2. Find the `index` of the Clifford object, translate it to canonical gates, and use the function `compose_gates`.

BTW, also when a Clifford is given as a list of gates we know it's a Clifford since it must use only the given Clifford gates (x,y,z,h,s,sdg,cx). 
So, I don't understand why it should be given as a Clifford object and not as Clifford gates.

Reply:
I agree we don't want to put too much effort into the `Clifford` class since it will get replaced by one in Terra, but I think these are quick fixes. Giving Clifford objects is more object-oriented and then we can do all error checking in the class.

Reply:
I discussed this with @chriseclectic .  He thinks that appending a compose method to the Clifford class is not immediate (since one has to be careful with updating the phases), and anyway this will be done in the new Clifford class in Terra.

So, we can either wait for the Terra Clifford class, or handle this for 1-2 qubits using an indirect way, namely: find the index of the Clifford object, translate it to canonical gates, and use the function compose_gates (this shouldn't be difficult).

Reply:
Ok, just pass in the list of gates...we'll wait for the Clifford class in terra (there will be a lot of updates for that anyway)